components/inputs/input-date.js

```diff
@@ -268,16 +271,16 @@ class InputDate extends LocalizeStaticMixin(LitElement) {
 		// open dropdown on down arrow or enter and focus on calendar focus date
 		if (e.keyCode === 40 || e.keyCode === 13) {
 			this._dropdown.open();
-			await this._handleChange(e);
+			await this._handleChange();
 			this._calendar.focus();
 			this._setFormattedValue();
 
 			if (e.keyCode === 40) e.preventDefault();
 		}
 	}
 
-	async _handleChange(e) {
-		const value = e.target.value;
+	async _handleChange() {
+		const value = this._textInput.value;
```

Comment:
Fixes the icon clearing the value (`mouseup` on `input-text` was triggered when `d2l-icon` was clicked but `icon` wouldn't have a value)